docs: Update features docs (#15122)
unoplatform/uno: doc/articles/features/windows-storage-pickers.md

# Storage Pickers

> [!TIP]

> > This article covers Uno-specific information for `Windows.Storage.Pickers` namespace. For a full description of the feature and instructions on using it, see [Windows.Storage.Pickers Namespace](https://learn.microsoft.com/uwp/api/windows.storage.pickers).

Reviewer: ```suggestion
> This article covers Uno-specific information for the `Windows.Storage.Pickers` namespace. For a full description of the feature and instructions on using it, see [Windows.Storage.Pickers Namespace](https://learn.microsoft.com/uwp/api/windows.storage.pickers).
```

| FileSavePicker | ✔    | ✔ (1)       | ✔      | ✔      | ✔    | ✔   | ✔  |
| FolderPicker   | ✔    | ✔           | ✔      | 💬 (2) | ✔    | ✖   | ✔  |

*(1) - Multiple implementations supported - see WebAssembly section below*\
*(2) - See iOS section below*

> On some platforms, you can further customize the file picking experience by utilizing additional properties:

Reviewer: ```suggestion
On some platforms, you can further customize the file-picking experience by utilizing additional properties:
```

| SuggestedStartLocation  | ✔   | ✔ (1)       | 💬     | ✔ (3) | ✔    | ✔   | ✔  |
| SettingsIdentifier      | ✔   | ✔ (1)       | ✔      | ✖     | ✖    | ✖   | ✖  |

> *(1) - Only for the native file pickers - see WebAssembly section below*\
> *(2) - For FileOpenPicker, VideosLibrary and PicturesLibrary are used apply `image/*` and `video/*` filters*\

Reviewer: ```suggestion
*(1) - Only for the native file pickers - see WebAssembly section below*\
*(2) - For FileOpenPicker, VideosLibrary and PicturesLibrary are used to apply `image/*` and `video/*` filters*\
```

}
```

> [!NOTE]

> > While the `SuggestedStartLocation` has currently no effect in Uno Platform targets, it must be set, otherwise the dialog crashes on UWP. `FileTypes` must include at least one item. You can add extensions in the format `.extension`, with the exception of `*` (asterisk) which allows picking any type of file.

Reviewer: ```suggestion
> While the `SuggestedStartLocation` has currently no effect on Uno Platform targets, it must be set, otherwise, the dialog crashes on UWP. `FileTypes` must include at least one item. You can add extensions in the format `.extension`, with the exception of `*` (asterisk) which allows picking any type of file.
```

}
```

> [!NOTE]

> > The `CachedFileManager` only has effect on Windows and on WebAssembly. For more information, see [WebAssembly section](#webassembly) below.

Reviewer: ```suggestion
> The `CachedFileManager` only affects Windows and WebAssembly. For more information, see [WebAssembly section](#webassembly) below.
```

>
> [!NOTE]
> While the `SuggestedStartLocation` has no effect, it must be set for UWP. You must declare at least one item for `FileTypeChoices`. Each has a description and one or more extensions.

## Picker configuration

File pickers have various configuration options that customize the experience. For a full list of properties, see the [UWP documentation](https://learn.microsoft.com/uwp/api/windows.storage.pickers.fileopenpicker). Not all options are supported on all target platforms, in which case these are ignored.

> To set which file type extensions you want to allow, use the `FileTypeFilter` property on `FileOpenPicker` and `FolderPicker`, and the `FileTypeChoices` property on `FileSavePicker`. Extensions must be in the format ".xyz" (starting with a dot). For `FileOpenPicker` and `FolderPicker` you can also include "*" (star) entry, which represents the fact that any file extension is allowed.

Reviewer: ```suggestion
To set which file type extensions you want to allow, use the `FileTypeFilter` property on `FileOpenPicker` and `FolderPicker`, and the `FileTypeChoices` property on `FileSavePicker`. Extensions must be in the format ".xyz" (starting with a dot). For `FileOpenPicker` and `FolderPicker` you can also include a "*" (star) entry, which represents the fact that any file extension is allowed.
```

The most powerful picker implementation on WebAssembly uses the <a href="https://wicg.github.io/file-system-access/" target="_blank">**File System Access API**</a>. This is not yet widely implemented across all browsers. See the following support tables for each picker:

* [`FolderPicker`](https://caniuse.com/?search=showDirectoryPicker)
* [`FileOpenPicker`](https://caniuse.com/?search=showOpenFilePicker)
* [`FileSavePicker`](https://caniuse.com/?search=showSaveFilePicker)

> `FolderPicker` is only supported for this type of pickers.

Reviewer: ```suggestion
`FolderPicker` is only supported for this type of picker.
``` 
Or ?
```suggestion
`FolderPicker` is only supported for these types of pickers.
```
